Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm

```diff
@@ -393,8 +393,13 @@ static bool ensureSandboxCacheDirectory(const SandboxInfo& info)
 
 static bool writeSandboxDataToCacheFile(const SandboxInfo& info, const Vector<char>& cacheFile)
 {
-    FileHandle file { info.filePath, FileSystem::FileOpenMode::Truncate, FileSystem::FileLockMode::Exclusive };
-    return file.write(cacheFile.data(), cacheFile.size()) == safeCast<int>(cacheFile.size());
+    // To avoid locking, write the sandbox data to a temporary path including the current process' PID
+    // then rename it to the final cache path.
+    auto temporaryPath = makeString(info.filePath, '-', getpid());
+    FileHandle file { temporaryPath, FileSystem::FileOpenMode::Truncate };
+    if (file.write(cacheFile.data(), cacheFile.size()) != safeCast<int>(cacheFile.size()))
```

Comment:
Should we clean up the temp file? I don't believe it automatically happens.

Reply:
Sounds good. I am now deleting the temporary file in this case.